clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -449,16 +465,29 @@ private static int parseAcks(String acksString) {
             if (callback != null)
                 callback.onCompletion(null, e);
             this.errors.record();
+            if (this.interceptors != null)
+                this.interceptors.onAcknowledgement(null, e);
             return new FutureFailure(e);
         } catch (InterruptedException e) {
             this.errors.record();
+            if (this.interceptors != null)
+                this.interceptors.onAcknowledgement(null, e);
             throw new InterruptException(e);
         } catch (BufferExhaustedException e) {
             this.errors.record();
             this.metrics.sensor("buffer-exhausted-records").record();
+            if (this.interceptors != null)
+                this.interceptors.onAcknowledgement(null, e);
             throw e;
         } catch (KafkaException e) {
             this.errors.record();
+            if (this.interceptors != null)
+                this.interceptors.onAcknowledgement(null, e);
+            throw e;
+        } catch (Exception e) {
```

Comment:
I am not sure why errors.record() was not called here before. If this is a bug, I can add the call to errors.record(). I did not want to change any other logic. Maybe it is better to open a bug for it and fix separately, including how we do catch here.